File: services/assessment-generation/app/utils/llm_client.py
import os
from typing import Optional

import requests


def query_llm(prompt: str, *, system_prompt: Optional[str] = None, model: Optional[str] = None, timeout_s: int = 120) -> str:
    """Call the repository's `services/llm` HTTP API and return raw text output."""
    base_url = (os.environ.get("LLM_SERVICE_URL") or "http://localhost:5003").rstrip("/")
    url = f"{base_url}/api/generate"
    payload = {"prompt": prompt}
    if system_prompt:
        payload["system_prompt"] = system_prompt
    if model:
        payload["model"] = model

    resp = requests.post(url, json=payload, timeout=timeout_s)
    try:
        data = resp.json()
    except Exception:
        resp.raise_for_status()
        raise

    if resp.status_code >= 400:
        raise ValueError(data.get("error") or "LLM service error")
    if "response" not in data:
        raise ValueError("LLM service returned no response")
    return data["response"]

File: backend/src/utils/llmClient.js
/**
 * LLM Service Client
 *
 * Wraps HTTP calls to the centralised Flask LLM micro-service (services/llm).
 * The service exposes POST /api/generate accepting { prompt, system_prompt }.
 */

import axios from "axios";

const LLM_SERVICE_URL = process.env.LLM_SERVICE_URL || "http://localhost:5003";
const LLM_REQUEST_TIMEOUT = parseInt(process.env.LLM_REQUEST_TIMEOUT || "120000", 10); // 120 s

const client = axios.create({
    baseURL: LLM_SERVICE_URL,
    timeout: LLM_REQUEST_TIMEOUT,
    headers: { "Content-Type": "application/json" },
});

/**
 * Send a prompt to the LLM service and return the generated text.
 *
 * @param {string} prompt        – The user/task prompt.
 * @param {string} [systemPrompt]– Optional system instruction.
 * @returns {Promise<string>}     – Raw text response from the LLM.
 */
export const llmGenerate = async (prompt, systemPrompt) => {
    const payload = { prompt };
    if (systemPrompt) payload.system_prompt = systemPrompt;

    try {
        const { data } = await client.post("/api/generate", payload);
        return data.response; // Flask service returns { response: "..." }
    } catch (error) {
        console.error("[LLM Client] Error:", error?.response?.data || error.message);
        throw new Error(
            `LLM service error: ${error?.response?.data?.error || error.message}`
        );
    }
};
